lib/oidcstrategy.js

```diff
@@ -69,18 +68,22 @@ const NONCE_MAX_AMOUNT = 10;
 const NONCE_LIFE_TIME = 3600; // second
 
 function makeProfileObject(src, raw) {
+  var emails = [];
+  if (src.upn)
+    emails = [src.upn];
```

Comment:
upn is not necessarily the email of the user. It might have `@` in it. I think this property should be called `username` instead of `emails`.